python_modules/dagster/dagster/_daemon/daemon.py

```diff
@@ -260,19 +262,46 @@ def core_loop(
 
 
 class SensorDaemon(DagsterDaemon):
+    def __init__(self) -> None:
+        super().__init__()
+        self._executor: Optional[ThreadPoolExecutor] = None
+        self._threadpool_executor: Optional[ThreadPoolExecutor] = None
+        self._exit_stack = ExitStack()
+
     @classmethod
     def daemon_type(cls) -> str:
         return "SENSOR"
 
+    def __exit__(self, _exception_type, _exception_value, _traceback):
+        self._exit_stack.close()
+        super().__exit__(_exception_type, _exception_value, _traceback)
     def core_loop(
         self,
         workspace_process_context: IWorkspaceProcessContext,
         shutdown_event: Event,
     ) -> DaemonIterator:
+        settings = workspace_process_context.instance.get_sensor_settings()
```

Comment:
Could also initialize the executors in `__init__` if we pass in the sensor settings there when we create the daemon. This way, we can keep `core_loop` clean
 https://github.com/dagster-io/dagster/blob/6509a4dfecd8d40ed65b38fd40e052e662d5b8ce/python_modules/dagster/dagster/_daemon/controller.py#L336-L340

Reply:
good idea